### PR TITLE
Support for ICC Parametric curves type 1-4

### PR DESCRIPTION
**Pull request by remia**
Address issue #1623 by supporting ICC Parametric Curve type 1-4 implemented as 1DLUT. I used a side-car python [script](https://gist.github.com/remia/4cab0fb18a794d699d8ae2d6b77091a4) during my tests to compare the results against pure Python or [DemoICCMax](https://github.com/InternationalColorConsortium/DemoIccMAX). Opening that now to allow time for discussion.

- Size of the 1DLUT is currently 1024, increasing that was decreasing the match against the pure Python implementation, I have a feeling this is related to inversion and flat slopes,
- Follows ICC [white paper](https://www.color.org/whitepapers/ICC_White_Paper35-Use_of_the_parametricCurveType.pdf) recommendations on parameters validation, there are no unit tests validating all possible failure case at the moment,
- Added a RangeTransform on the pure exponent path to follow ICC specs more closely. Unit tests sample updated using directly OCIO results for the case where clipping changed the expected values as I don't know exactly how the values were obtained,
- As before, all the TRC tags in the profile should be of the same type, the individual parameters can vary among red, green and blue but the function type should match.

### CODE REVIEW

**Review by michdolan — APPROVED** [review]

**Comment by michdolan** [comment]
This looks good to me, but I think @doug-walker should review before merging.

File: src/OpenColorIO/fileformats/FileFormatICC.cpp
Comment (doug-walker) [inline src/OpenColorIO/fileformats/FileFormatICC.cpp:373]: I notice in your Python script you implemented some additional checks (e.g. continuity at the break-point) that are warnings rather than errors.  Would it make sense to include those here and call LogWarning?

**Review by doug-walker — CHANGES_REQUESTED** [review]
This is awesome Remi, nice work!

File: src/OpenColorIO/fileformats/FileFormatICC.cpp
Comment (remia) [reply src/OpenColorIO/fileformats/FileFormatICC.cpp:421]: Yeah I noticed that too, I think it is now fixed in the latest ICC.1:2022 (Profile version 4.4.0.0) spec.

File: src/OpenColorIO/fileformats/FileFormatICC.cpp
Comment (remia) [reply src/OpenColorIO/fileformats/FileFormatICC.cpp:304]: Will fix the typos thanks.

File: tests/cpu/fileformats/FileFormatICC_tests.cpp
Comment (remia) [reply tests/cpu/fileformats/FileFormatICC_tests.cpp:517]: Good catch, I'll add a smaller value for better coverage and add the parameters for the curves as comments.

File: src/OpenColorIO/fileformats/FileFormatICC.cpp
Comment (remia) [reply src/OpenColorIO/fileformats/FileFormatICC.cpp:373]: I'll try to add them good point.

Comment (remia) [reply src/OpenColorIO/fileformats/FileFormatICC.cpp:464]: I added the warning from the Python script, note that I did bump the comparison tolerance by using 8 bits quantisation for these to avoid warnings on standard ICC profiles like Display P3. What do you think @doug-walker ?

Comment (doug-walker) [reply src/OpenColorIO/fileformats/FileFormatICC.cpp:464]: Thanks for adding!  Yes, I think the 8-bit quantization is appropriate to avoid unnecessary warnings.

**Review by doug-walker — APPROVED** [review]
All comments addressed, thank you Remi!

**Pull request merged**
